cmd/observe/flows_filter.go

```diff
@@ -108,8 +180,14 @@ type flowFilter struct {
 func newFlowFilter() *flowFilter {
 	return &flowFilter{
 		conflicts: [][]string{
-			{"from-fqdn", "from-ip", "from-namespace", "from-pod", "fqdn", "ip", "namespace", "pod"},
-			{"to-fqdn", "to-ip", "to-namespace", "to-pod", "fqdn", "ip", "namespace", "pod"},
+			{"from-fqdn", "from-ip", "ip", "fqdn", "from-namespace", "namespace"},
+			{"from-fqdn", "from-ip", "ip", "fqdn", "from-pod", "pod"},
+			{"to-fqdn", "to-ip", "ip", "fqdn", "to-namespace", "namespace"},
+			{"to-fqdn", "to-ip", "ip", "fqdn", "to-pod", "pod"},
+			{"to-pod", "namespace"},
+			{"from-pod", "namespace"},
+			{"to-service", "namespace"},
+			{"from-service", "namespace"},
```

Comment:
Seems to me that the patch doesn't introduce new incompatibilities, but lift of some (e.g. before the patch `--pod` and `--namespace` are incompatible, but ok after the patch). The [conflict test case coverage](https://github.com/cilium/hubble/blob/f65467be66e7db072e14ca202d2d7b1f4d56b5df/cmd/observe/flows_filter_test.go#L42-L56) is pretty low, so if correct it could be good to add a case for `namespace` and `pod`? If incorrect and new incompatibilities are added we should label the PR with `breaking-change`.

Reply:
There should not be any new incompatibilities, but I agree the the test coverage for this is very low and it's starting to get fairly complicated. I'll add a few test cases to be sure that this works as expected.

Reply:
I added tests and realized we do add a new incompatibility. `hubble observe --namespace foo --to-service/--from-service bar` isn't valid anymore.

Reply:
Ok, marking the PR as a breaking change, please state this incompatibility in the PR's description so it will be easier to find back when making the releases notes :pray: